File: Napakalaki-Ruby/lib/combat_result.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module NapakalakiGame

module CombatResult
    WINGAME=:wingame
    WIN=:win
    LOSE=:lose
    LOSEANDCONVERT=:loseandconvert
    
   def to_s
    out = case
      when self == WINGAME then "Eres el vencedor :D\n"
      when self == WIN then "¡Ganaste! \n"
      when self == LOSE then "¡Oh! Perdiste... \n"
      when self == LOSEANDCONVERT then "Has perdido, y ahora eres sectario.\n" 
      end
  
    out
  end
end
end

File: Napakalaki-Java/src/NapakalakiGame/CombatResult.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author Laura
 */
public enum CombatResult {
    WINGAME, WIN, LOSE, LOSEANDCONVERT;
    public String toString(){
        String out=null;
        switch(this){
            case WIN:
                out= "¡Ganaste!";
                break;
            case LOSE:
                out= "¡Oh! Perdiste...";
                break;
            case LOSEANDCONVERT:
                out= "Perdiste y te convertiste.";
            case WINGAME:
                out= "Eres el vencedor :D";
                break;
        }
        return out;
    }
}
